### DISCUSSION TITLE
Moniker for net core frameworks

### DISCUSSION THREAD

**Discussion by axunonb** [SteveGilham/altcover#232] [Q&A] [answered]
Were using the following command to create the coverage xml:
`dotnet test Solution.sln`--framework net60 --test-adapter-path:. --logger:Appveyor /p:configuration=release /p:AltCover=true /p:AltCoverStrongNameKey="..\Solution.snk" /p:AltCoverAssemblyExcludeFilter="Solution.Tests|NUnit3.TestAdapter" /p:AltCoverAttributeFilter="ExcludeFromCodeCoverage" /p:AltCoverLineCover="true"`

Using `--framework net6.0` as the (correct) moniker we see no output files, while `--framework net60` does produce `coverage.net60.xml`.
What are we missing here?

**Comment by SteveGilham** [comment, marked as answer]
The infill in the `coverage.[whatever].xml` is taken from the [`$(TargetFramework)`](https://github.com/SteveGilham/altcover/blob/79af67ee9b8f6ba42dbaa08558c7fa3ab28a6024/nupkg/build/AltCover.proj#L40-L42) value, so is driven by the `--framework` value on the command line.

So, for example, from the most recent (at time of writing) build at line 7669
`D:\a\altcover\altcover\__AltCover.Api.Tests> "C:\Program Files\dotnet\dotnet.EXE"  test --verbosity minimal AltCover.Api.Tests.fsproj --configuration Debug --framework net9.0 /nodeReuse:False /p:AltCoverTag=UnitTestWithCoreRunner_ /p:AltCover=true /p:AltCoverAssemblyFilter=%5C.DataCollector|Sample|Microsoft|testhost| /p:AltCoverTypeFilter=System%5C.|Sample3%5C.Class2|Microsoft|<Start|ICSharpCode|UnitTestStub|SolutionRoot|Expecto| /p:AltCoverMethodFilter=WaitForExitCustom| /p:AltCoverAttributeFilter=TestSDKAutoGeneratedCode| /p:AltCoverStrongNameKey=D:%5Ca%5Caltcover%5Caltcover%5CBuild%5CSelfTest.snk /p:AltCoverReport=D:%5Ca%5Caltcover%5Caltcover%5C_Reports%5CApiTestWithAltCoverCoreRunner.xml /p:AltCoverReportFormat=OpenCover /p:AltCoverShowStatic=- /p:AltCoverAll=true /p:AltCoverLocalSource=true /p:AltCoverVisibleBranches=true /p:AltCoverShowSummary=Green /p:AltCoverForce=true (In: false, Out: false, Err: false)`

with `--framework net9.0` included, yields

```
...
     => D:\a\altcover\altcover\_Binaries\UnitTestWithCoreRunner_AltCover.Api.Tests\Debug+AnyCPU\net9.0\AltCover.Toolkit.dll
  
  Coverage Report: D:\a\altcover\altcover\_Reports\ApiTestWithAltCoverCoreRunner.net9.0.xml
  
  
      D:\a\altcover\altcover\_Binaries\UnitTestWithCoreRunner_AltCover.Api.Tests\Debug+AnyCPU\net9.0\__Instrumented_AltCover.Api.Tests\AltCover.Base.dll
                  <=  AltCover.Base, Version=9.0.0.0, Culture=neutral, PublicKeyToken=3eb3cd726036ff71
...
```

Have you checked the `<TargetFrameworks>` clause in your test project file?

**Reply by axunonb** [reply]
Thanks for the answer, especially for that last line. This was the cause of the problem.
